src/rgw/rgw_sal_motr.cc

```diff
@@ -1511,26 +1509,24 @@ void MotrObject::read_bucket_info(const DoutPrefixProvider* dpp, std::string& bn
 }
 
 int MotrObject::modify_obj_attrs(RGWObjectCtx* rctx, const char* attr_name, bufferlist& attr_val, optional_yield y, const DoutPrefixProvider* dpp)
-{
+{  
   rgw_obj target = get_obj();
-  int r = get_obj_attrs(rctx, y, dpp, &target);
```

Comment:
if no attributes exists, then yes it doesn't make sense to proceed, given function is to modify any existing attributes.
so this code needed still

Reply:
attr_name and attr_val will now come through arguments, put into a sal::Attrs type and will be passed to modify_obj_attrs method.
Since we are not directly dealing with attrs member object now, there is no role of get_obj_attrs here.

Reply:
> attr_name and attr_val will now come through arguments

they are already coming as arguments!

Even before calling set or delete obj attrs, this get call will help in case modify cant be taken up.so this code needed still

Reply:
I understand we should first get the attrs and then set them with updated values.
get_obj_attrs is calling nothing but fetch_obj_entry_and_key, which is already called by set_obj_attrs in the beginning. hence as per my understanding `get_obj_attrs(rctx, y, dpp, &target)` call will be redundant here..

Reply:
got it. ideally we need to differentiate between rc, whether its because of non-existent attrs or failed to fetch it in fetch_obj_entry_and_key().